server/controllers/user.controller.js

```diff
@@ -144,6 +143,37 @@ class UserController {
       Response(res, 400, errorMessage);
     }
   }
+
+  /**
+     * @description updates the role of a user
+     * @param {object} req
+     * @param {object} res
+     * @returns {object} User Profile Object
+     */
+  static async updateUserRole(req, res) {
+    const { id } = req.params;
+    const { newrole } = req.body;
+    const foundUser = await User.findByPk(id);
+
+    if (!foundUser) return Response(res, 404, 'This user was not found.');
+
+    const newRoleId = await Role.findOne({ where: { type: newrole } });
+    const updatedUser = await User.update({ roleId: newRoleId.dataValues.id }, {
+      returning: true,
+      where: { id },
+    });
+
+    const { dataValues } = updatedUser[1][0];
```

Comment:
Destructure dataValues further. You would have something like this: 
```
const { dataValues: { id, firstname, lastname, newrole, updatedAt } } = updatedUser[1][0];
const result = {
  id, 
  firstname,
  lastname,
  role: newrole,
  updatedAt
}```
Implement this here and in every other place necessary

Reply:
Fixed it. Thanks for the example.